art/attacks/evasion/over_the_air_flickering/over_the_air_flickering_pytorch.py

```diff
@@ -0,0 +1,360 @@
+from art.attacks.attack import EvasionAttack
+from art.utils import check_and_transform_label_format, get_labels_np_array
+
+if TYPE_CHECKING:
+    # pylint: disable=C0412
+    import torch
+    from art.utils import CLASSIFIER_LOSS_GRADIENTS_TYPE
+
+
+logger = logging.getLogger(__name__)
+
+
+class OverTheAirFlickeringPyTorch(EvasionAttack):
+    """
+    This module contains an implementation of the Over-the-Air Adversarial Flickering attack on video recognition
+    networks.
+
+    | Paper link: https://arxiv.org/abs/2002.05123
+    """
+
+    attack_params = EvasionAttack.attack_params + [
+        "eps_step",
+        "max_iter",
+        "beta_0",
+        "beta_1",
+        "beta_2",
+        "loss_margin",
+        "batch_size",
+        "start_frame_index",
+        "num_frames",
+        "round_samples",
+    ]
+
+    _estimator_requirements = (BaseEstimator, ClassifierMixin, LossGradientsMixin)
+
+    def __init__(
+        self,
+        classifier: "CLASSIFIER_LOSS_GRADIENTS_TYPE",
+        eps_step: float = 0.01,
+        max_iter: int = 30,
+        beta_0: float = 1.0,
+        beta_1: float = 0.5,
+        beta_2: float = 0.5,
+        loss_margin: float = 0.05,
+        batch_size: int = 1,
+        start_frame_index: int = 0,
+        num_frames: Optional[int] = None,
+        round_samples: float = 0.0,
+    ):
+        """
+        Create an instance of the :class:`.OverTheAirFlickeringPyTorch`.
+
+        :param classifier: A trained classifier.
+        :param eps_step: The step size per iteration.
+        :param max_iter: The number of iterations.
+        :param beta_0: Weighting of the sum of all regularisation terms corresponding to `lambda` in the original paper.
+        :param beta_1: Weighting of thickness regularisation.
+        :param beta_2: Weighting of roughness regularisation.
+        :param loss_margin: The loss margin.
+        :param batch_size: Batch size.
+        :param start_frame_index: The first frame to be perturbed.
+        :param num_frames: The number of frames to be perturbed.
+        :param round_samples: Granularity of the input values to be enforced if > 0.0.
+        """
+        super().__init__(estimator=classifier)
+
+        self.eps_step = eps_step
+        self.max_iter = max_iter
+        self.beta_0 = beta_0
+        self.beta_1 = beta_1
+        self.beta_2 = beta_2
+        self.loss_margin = loss_margin
+        self.batch_size = batch_size
+        self.start_frame_index = start_frame_index
+        self.num_frames = num_frames
+        self.round_samples = round_samples
+        self.end_frame_index = (
+            self.start_frame_index + self.num_frames if self.num_frames is not None else self.num_frames
+        )
+        self._check_params()
+
+    def generate(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> np.ndarray:
+        """
+        Generate adversarial examples.
+
+        :param x: Original input samples.
```

Comment:
It is worth to specify the shape of `x` here because `x` now is not an image.

Reply:
I agree.